dsymprocessor/logs_processor.go

```diff
@@ -61,17 +63,139 @@ func (sp *symbolicatorProcessor) processResourceSpans(ctx context.Context, rl pl
+
+
+	var ok bool
+	var stackTraceValue pcommon.Value
+	var binaryNameValue pcommon.Value
+	var buildUUIDValue pcommon.Value
+
+	if stackTraceValue, ok = attributes.Get(sp.cfg.StackTraceAttributeKey); !ok {
+		// we should never get here (our caller checks this)
+		return fmt.Errorf("Invalid state! Called proceStackTraceAttributes while missing %s attribute", sp.cfg.StackTraceAttributeKey)
+	}
+	rawStackTrace := stackTraceValue.Str()
+	
+	if buildUUIDValue, ok = attributes.Get(sp.cfg.BuildUUIDAttributeKey); !ok {
+		return fmt.Errorf("%w: %s", errMissingAttribute, sp.cfg.BuildUUIDAttributeKey)
+	}
+	buildUUID := buildUUIDValue.Str()
+
+	if binaryNameValue, ok = attributes.Get(sp.cfg.AppExecutableAttributeKey); !ok {
+		return fmt.Errorf("%w: %s", errMissingAttribute, sp.cfg.AppExecutableAttributeKey)
+	}
+	binaryName := binaryNameValue.Str()
+
+	lines := strings.Split(rawStackTrace, "\n")
+	res := make([]string, len(lines))
+	for idx,line := range(lines) {
+		symbolicated,err := sp.symbolicateStackLine(ctx, line, binaryName, buildUUID)
+		if (err != nil) {
+			sp.logger.Debug("could not symbolicate line")
+			res[idx] = line
+			continue
+		}
+		res[idx] = symbolicated
+	}
+
+	if sp.cfg.PreserveStackTrace {
+		attributes.PutStr(sp.cfg.OriginalStackTraceKey, rawStackTrace)
+	}
+	attributes.PutStr(sp.cfg.StackTraceAttributeKey, strings.Join(res, "\n"))
+	attributes.PutBool(sp.cfg.SymbolicatorFailureAttributeKey, false)
+
+	return nil
+}
+
+func (sp *symbolicatorProcessor) symbolicateStackLine(ctx context.Context, line, binaryName, buildUUID string) (string, error) {
+	// groups: line number, library name, hex address, uuid or binary name, offset
+	stackLineRegex := regexp.MustCompile(`^([0-9]+)\s+([\w _\-\.]+[\w_\-\.])\s+(0x[\da-f]+)\s+([\w _\-\.]*) \+ (\d+)`)
+	matches := stackLineRegex.FindStringSubmatch(line)
+	matchIdxes := stackLineRegex.FindStringSubmatchIndex(line)
+	libName := matches[2]
+	uuidOrBinary := matches[4]
+	offsetInt,err := strconv.Atoi(matches[5])
+	if err != nil {
+		return "", err
+	}
+	offset := uint64(offsetInt)
+
+	var uuid string
+	var bin string
+	if isUUID(uuidOrBinary) {
+		uuid = uuidOrBinary
+		bin = libName
+	} else if uuidOrBinary == binaryName {
+		uuid = buildUUID
+		bin = binaryName
+	} else {
+		return line, nil
+	}
+
+	locations, err := sp.symbolicator.symbolicateFrame(ctx, uuid, bin, offset)
+
+	if errors.Is(err, errFailedToFindDSYM) {
+		return line, nil
+	}
+	if err != nil {
+		return "", err
+	}
+	
+	// keep everything up to the end of match group 3 (the binary/uuid)
+	//   indexes are paired, so group 0 spans index 0 - index 1
+	//   so index 7 is the end of group 3
+	prefix := line[:matchIdxes[7]]
+
+	return formatStackFrames(prefix, bin, offset, locations), nil
+}
+
+func isUUID (maybeUUID string) bool {
+	uuidRegex := regexp.MustCompile(`[0-9A-Z]{8}-[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{4}-[0-9A-Z]{12}`)
+	return uuidRegex.Match([]byte(maybeUUID))
```

Comment:
you can do match string to avoid the cast

```suggestion
	return uuidRegex.MatchString(maybeUUID)
```